### PR TITLE
separate base and utility

### PR DESCRIPTION
**Pull request by mikecovlee**

### CODE REVIEW

File: mozart++/core/base.hpp [hunk 1]
```diff
@@ -79,7 +117,28 @@ namespace mozart {
...
+// Function importing
+    using std::declval;
+    using std::forward;
+    using std::memcpy;
+    using std::move;
+    using std::swap;
+// Alignment
+    template<typename type> using aligned_type = typename std::aligned_storage<sizeof(type), std::alignment_of<type>::value>::type;
+    template<size_t len, typename ...types> using aligned_union = typename std::aligned_union<len, types...>::type;
+    inline byte_t *uninitialized_copy(byte_t *, byte_t *, size_t) noexcept;
```
Comment (imkiva) [inline mozart++/core/base.hpp]: Insert some empty lines

File: mozart++/core/base.hpp [hunk 2]
```diff
@@ -79,7 +117,28 @@ namespace mozart {
...
+    template<typename T, typename ...Args>
+    inline static void construct_at(byte_t *ptr, Args &&...args)
+    {
+        ::new (ptr) T(forward<Args>(args)...);
+    }
+    template<typename T>
```
Comment (imkiva) [inline mozart++/core/base.hpp]: ditto

File: mozart++/core/utility.hpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        enum class time_unit {
+            nano_sec,
+            micro_sec,
+            milli_sec,
+            second,
+            minute
+        };
+
+        static void reset()
+        {
```
Comment (imkiva) [inline mozart++/core/utility.hpp]: Why this bracket indent differs from the previous ones???

File: mozart++/core/utility.hpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+namespace mozart {
+    class timer final {
+        static std::chrono::time_point<std::chrono::high_resolution_clock> m_timer;
+
+    public:
+        enum class time_unit {
+            nano_sec,
+            micro_sec,
+            milli_sec,
```
Comment (imkiva) [inline mozart++/core/utility.hpp]: Consider `nanoseconds`, `microseconds`, `milliseconds`, `seconds` and `minutes`

File: src/mozart.cpp [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+#include <mozart++/mozart>
+
+mozart::byte_t *mozart::uninitialized_copy(byte_t *dest, byte_t *src, size_t count) noexcept
+{
+    return reinterpret_cast<byte_t *>(memcpy(reinterpret_cast<void *>(dest), reinterpret_cast<void *>(src), count));
+}
+
+std::chrono::time_point<std::chrono::high_resolution_clock> mozart::timer::m_timer(std::chrono::high_resolution_clock::now());
```
Comment (imkiva) [inline src/mozart.cpp]: Add a line break if a line is too long to be displayed in a screen.

File: mozart++/core/utility.hpp (hunk 3 above)
Comment (mikecovlee) [reply mozart++/core/utility.hpp]: just follow formatter

**Pull request merged**